packages/cli/src/lib/implementation/core-config.middleware.integration.test.ts

```diff
@@ -35,3 +51,73 @@ describe('coreConfigMiddleware', () => {
     ).rejects.toThrow(/Provided path .* is not valid./);
   });
 });
+
+const cliWithConfigOptionsAndMiddleware = (
+  cliObj: Partial<GeneralCliOptions>,
+) =>
+  yargsCli<CoreConfig>(objectToCliArgs(cliObj), {
+    options: {
+      ...yargsCoreConfigOptionsDefinition(),
+    },
+    middlewares: [{ middlewareFunction: coreConfigMiddleware }],
+  });
+
+describe('cliWithConfigOptionsAndMiddleware', () => {
```

Comment:
For better test readability and maintainability, one should be able to read the test from top to bottom and understand what it does without scrolling around or investigating other files.

This means:
- clear use case name to understand the test purpose
- minimum helper functions or variables
- clear, explicit usage of values passed to the test and expected result values

Such a test could look e.g. like this (flow taken from the last test case):

```
  it('should return persist filename from config, outputDir from CLI and format default', async () => {
    await expect(
      yargsCli<CoreConfig>(
        [
          '--persist.outputDir=tmp-cli',
          `--config=${join(
            configDirPath,
            'code-pushup.persist-only-filename.config.ts',
          )}`,
        ],
        {
          options: { ...yargsCoreConfigOptionsDefinition() },
          middlewares: [{ middlewareFunction: coreConfigMiddleware }],
        },
      ).parseAsync(),
    ).resolves.toEqual(
      expect.objectContaining({
        persist: {
          outputDir: 'tmp-cli',
          format: ['json'],
          filename: 'rc-report',
        },
      }),
    );
  });
  ```

Reply:
adopted the test name.